src/bin/pg_autoctl/file_utils.c

```diff
@@ -518,3 +671,39 @@ set_program_absolute_path(char *program, int size)
 
 	return true;
 }
+
+
+/*
+ * normalize_filename returns the real path of a given filename, resolving
+ * symlinks and pruning double-slashes and other weird constructs.
```

Comment:
It is worth adding a note that if the file does not exists, the same path is returned without any normalization attempt, or something like that.